tests/torch/test_nncf_network.py

```diff
@@ -1090,3 +1090,29 @@ def wrapper(*args, **kwargs):
 
     # Ideally, the condition below should fail but currently there is no means to implement it
     assert inspect.signature(original_model.forward) != inspect.signature(nncf_model.forward)
+
+
+def test_forward_hooks_are_preserved():
+    # Testing only for the forward hook - other hooks use the same mechanism.
+    original_obj = SimplestModel()
+
+    class CallCounter:
+        def __init__(self):
+            self.enabled = True
+            self.call_count = 0
+
+        def __call__(self, *args, **kwargs):
+            if self.enabled:
+                self.call_count += 1
+
+    hook = CallCounter()
+    original_obj.register_forward_hook(hook)
+
+    hook.enabled = False
+    nncf_net = NNCFNetwork(original_obj, [ModelInputInfo(SimplestModel.INPUT_SIZE)])
+    hook.enabled = True
+
+    assert len(nncf_net._forward_hooks) == 1
+    assert next(iter(nncf_net._forward_hooks.values())) is hook
+    nncf_net(torch.ones(SimplestModel.INPUT_SIZE))
+    assert hook.call_count == 1
```

Comment:
What was before the fix? Was count equal 0?

Reply:
Indeed.